modular_skyrat/modules/contractor/code/items/baton.dm

```diff
@@ -52,24 +69,36 @@
 
 /obj/item/melee/baton/telescopic/contractor_baton/additional_effects_non_cyborg(mob/living/carbon/target, mob/living/user)
 	. = ..()
-	if(!((upgrade_flags & BATON_MUTE_UPGRADE) && !(upgrade_flags & BATON_ALL_UPGRADE))|| !istype(target))
+	if(!istype(target))
 		return
-	if(target.silent >= (MUTE_CYCLES * 2))
-		return
-	target.silent += MUTE_CYCLES //10 seconds of mute a hit up to 20 max
+	if((upgrade_flags & BATON_MUTE_UPGRADE) || (upgrade_flags & BATON_ALL_UPGRADE))
+		if(target.silent < (MUTE_CYCLES * MUTE_MAX_MOD))
+			target.silent += MUTE_CYCLES //10 seconds of mute a hit up to 20 max
```

Comment:
If you want it to be a max of 20 seconds, you should clamp that value, because if they hit 19 seconds, they can go for another hit, and it'll bring the value to 29 seconds.

Reply:
done